pulsar-websocket/src/main/java/com/yahoo/pulsar/websocket/AbstractWebSocketHandler.java

```diff
@@ -81,10 +81,14 @@ public void onWebSocketConnect(Session session) {
                     log.warn("[{}] WebSocket Client [{}] is not authorized on topic {}", session.getRemoteAddress(), role,
                             topic);
                     close(WebSocketError.NotAuthorizedError);
+                } else {
+                    createClient(session);
```

Comment:
> onWebSocketText(..) gets triggered  only when server releases thread from onWebSocketConnect()

It means if we do authorization async then there could be a chance that producer start producing messages even before authorization completed. So, we can keep your previous commit where we do authorization synchronously and it should be faster because we fetch it from ZkCache and get from zk only once.